maven-plugins/enforcer-maven-plugin/src/main/java/io/helidon/build/maven/enforcer/inclusivenaming/InclusiveNamingRule.java

```diff
@@ -0,0 +1,246 @@
+                    }
+                }
+                if (!found.isEmpty()) {
+                    errors.add(RuleFailure.create(fr, lineNum, "inclusive naming found: " + String.join(", ", found.toString())));
+                }
+                found.clear();
+            }
+
+        } catch (IOException e) {
+            throw new EnforcerException("Failed to process file for inclusive naming: " + fr.relativePath());
+        }
+    }
+
+    private boolean include(FileRequest fr) {
+        for (FileMatcher exclude : excludes) {
+            if (exclude.matches(fr)) {
+                return false;
+            }
+        }
+
+        for (FileMatcher include : includes) {
+            if (include.matches(fr)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * Fluent API builder for {@link io.helidon.build.maven.enforcer.inclusivenaming.InclusiveNamingRule}.
+     */
+    public static class Builder {
+        private InlusiveNamingConfig inclusiveNamingConfig;
+
+        private Builder() {
+        }
+
+        /**
+         * Update builder from maven configuration.
+         *
+         * @param inclusiveNamingConfig maven config
+         * @return updated builder
+         */
+        public Builder config(InlusiveNamingConfig inclusiveNamingConfig) {
+            this.inclusiveNamingConfig = inclusiveNamingConfig;
+            return this;
+        }
+
+        /**
+         * Build a new instance of {@link io.helidon.build.maven.enforcer.inclusivenaming.InclusiveNamingRule}.
+         *
+         * @return new rule
+         */
+        public InclusiveNamingRule build() {
+            return new InclusiveNamingRule(this);
+        }
+
+        private List<Term> readTerms(String file){
+            List<Term> terms = new ArrayList<>();
+            try (InputStream is = InclusiveNamingRule.class.getResourceAsStream(file)) {
+                XmlInclusiveNaming xml = xmlInclusiveNaming(is);
+                for (XmlData data : xml.getData()) {
+                    String tier = data.getTier();
+                    String term = data.getTerm();
+                    if (!ALLOWED_TIERS.contains(tier) && !inclusiveNamingConfig.excludeTerms().contains(term)) {
+                        String recommendation = data.getRecommendation();
+                        String termPage = data.getTermPage();
+                        terms.add(new Term(term, tier, recommendation, termPage,
+                                Arrays.asList(data.getRecommendedReplacements())));
+                    }
+                }
+            } catch (IOException | JAXBException e) {
+                throw new EnforcerException("Failed to read inclusive naming file", e);
+            }
+            return terms;
+        }
+
+        private XmlInclusiveNaming xmlInclusiveNaming(InputStream is) throws JAXBException {
+            JAXBContext contextObj = JAXBContext.newInstance(XmlInclusiveNaming.class);
```

Comment:
It is preferable to allow for terms data configuration inline inside a pom.xml to allow for additional terms, or just fully defined in a pom file.

You might be able to support external file and inline configuration if the beans are annotated with both JAXB annotations and Maven/Plexus annotations.

You may also be able to use Maven/Plexus annotations only, however reading from an external file will require a bit of digging (I would start here: https://github.com/eclipse/sisu.plexus/tree/master/org.eclipse.sisu.plexus)

You can also take some inspiration from the stager plugin that uses `PlexusConfiguration` but that's a DOM like API.

One advantage of doing binding with plexus is that we get a pseudo schema out of the box and most IDE provide completion for it in pom files.

Reply:
The names are not customized with the `@Parameter` annotations, e.g. `term_page`, does it convert camel case to `_` separated ?

Can you add an integration test (see how it's done with other plugins) to exercise and show-case the Maven configuration.

Reply:
I was fighting some time with the 'name' and the 'alias' of the Parameter, and I was not able to keep the class fields without underscore. I tried:
```
    @Parameter(name = "term_page", alias = "termPage")
    private String termPage = "";
```

and:

```
    @Parameter(name = "termPage", alias = "term_page")
    private String termPage = "";
```
None of them works. When you set in the plugin `<term_page>whatever</term_page>` it will complain saying that there is no term_page field in that class.

So I decided to not make my life more difficult. As we are processing the JSON anyway, I changed `<term_page>` by `<termPage>` and `<recommended_replacements>` by `<recommendedReplacements>`.

Reply:
Integration test added too